### PR TITLE
.github: add tests to CI

### PR DESCRIPTION
**Pull request by crawshaw**

### CODE REVIEW

File: .github/workflows/test-sqlite.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: test-sqlite
+
+on:
+  push:
+    branches: [ $default-branch ]
```
Comment (creachadair) [inline .github/workflows/test-sqlite.yml:5]: Does this work here? I thought this was only for templates.

Do we need `${{ github.event.repository.default_branch }}` ?

Comment (crawshaw) [reply .github/workflows/test-sqlite.yml]: ooh it worked this time!

lmk if you have a shorter form of this.

File: .github/workflows/test-sqlite.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: test-sqlite
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
Comment (creachadair) [inline .github/workflows/test-sqlite.yml:7]: Consider maybe also
```yml
types: [opened, reopened, synchronize]
```

Comment (crawshaw) [reply .github/workflows/test-sqlite.yml:7]: what does that do? (it seems to re-run on pushing to a PR already)

File: .github/workflows/test-sqlite.yml (hunk 1 above)
Comment (creachadair) [reply .github/workflows/test-sqlite.yml]: Yeah, I mean I just put in the specific name. It's not like default branch names change all that often anyway.

File: .github/workflows/test-sqlite.yml (hunk 2 above)
Comment (creachadair) [reply .github/workflows/test-sqlite.yml:7]: To be more specific:
```yml
on:
  push:
    branches:
      - main
  pull_request:
    types: [opened, reopened, synchronize]
```

(syntax mutatis mutandis)

File: .github/workflows/test-sqlite.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (creachadair) [inline .github/workflows/test-sqlite.yml:8]:
```suggestion
permissions:
  contents: read

```

though it is possible that is the default now.

Comment (crawshaw) [reply .github/workflows/test-sqlite.yml:8]: I think it is. Going to skip this on the principle that fewer lines is better.

Comment (creachadair) [reply .github/workflows/test-sqlite.yml:8]: It looks like the default is `read-all`, which is more general (e.g., it allows access to readable properties of the repo besides the files), but that is probably not a big issue.

File: .github/workflows/test-sqlite.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.21'
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run: go test -v ./...
```
Comment (creachadair) [inline .github/workflows/test-sqlite.yml:24]: `-race` ? 😄

**Review by creachadair — APPROVED** [review]

**Comment by raggi** [comment]
@creachadair i bumped this, added race as you suggested, LGTY?

**Review by creachadair — APPROVED** [review]
I had forgotten about the existence of this PR, but yes, still LGTM. 😆

**Pull request merged**